index.d.ts

```diff
@@ -135,6 +135,23 @@ export function constructor<T>(
   constructor: Constructor<T>
 ): TestDoubleConstructor<T>;
 
+//
+// fake: instance objects
+//
+
+/**
+ * Construct an instance of a faked class.
+ *
+ * @export
+ * @template T
+ * @param {{ new (...args: any[]): T }} constructor
+ * @returns {DoubledObject<typeof T>}
```

Comment:
Was not sure what the return type should be here

Reply:
@lgandecki as resident typescript typings maintainer could you please help with this / add a suggestion for what it should be?

Reply:
Sure thing. Looks correct. I've also verified it by cloning the branch and playing around with it. 
Thanks for your work @woldie .
I guess you could possibly get similar results using td.object(Class), but that uses proxy.. and is a bit less safe if used without the additional help of typescript

Reply:
Thanks @lgandecki!  I agree it would be more convenient to have `td.object(Class)` just do it all, but I think it's probably important to cordon off this particular convenience method since it is only a wrapper for a call `td.constructor`, but then drops the constructor on the floor when it is done with it, intentionally hiding the ability to mock static class methods.

Looking at this particular change again, I think my use of `typeof T` is in the wrong spot for what I want here.  I use JSDoc to specify types in my home project and I cribbed from it, but I think I cribbed without checking my work.  `typeof T` describes a constructor of type `T` when used in conjunction with `T` by itself which is meant to indicate an instance of `T`.  When I do a good job with my JSDoc, it works just as well as TypeScript for most things in terms of the quality of IntelliSense and inspections within IntelliJ.

In my home project, I have the need to annotate a constructor, and I needed IntelliJ to understand that the return value from the annotation function was the constructor.  Here's how I expressed that in JSDoc:

```
/**
 * @param {typeof T} ctor constructor function.
 * @param {...(number|string)} args metadata about ctor.
 * @returns {typeof T} annotated constructor function
 * @template T
 */
Injector.prototype.annotateConstructor = function (ctor, args) {
  ...
}
```

If the JSDoc part of a TypeScript declaration understands `typeof T` the same way, then the TypeScript type declaration should read:

```
/**
 * Construct an instance of a faked class.
 *
 * @export
 * @template T
 * @param {typeof T} constructor
 * @returns {DoubledObject<T>}
 */
export function instance<T>(
  constructor: Constructor<T>
): DoubledObject<T>;
```

I'll try this out later today and see how IntelliJ interprets it.

Reply:
Sounds good. I'm unfortunately not really familiar with JSDocs :-(